[nethost] Update to 7.0.0

julianxhokaxhiu opened the pull request:
Update the nethost port to v7.0.0 following the latest .NET 7 release announcement: https://devblogs.microsoft.com/dotnet/announcing-dotnet-7/

- #### What does your PR fix?
N/A

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
Same as previous version

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Yes, I've used `vcpkg x-add-version nethost`

julianxhokaxhiu committed:
[nethost] Update to 7.0.0

julianxhokaxhiu commented:
@microsoft-github-policy-service agree

Adela0814 commented on ports/nethost/0001-nethost-cmakelists.patch:
```diff
@@ -1,73 +1,78 @@
++    DESTINATION share/unofficial-nethost
++    FILE unofficial-nethost-config.cmake
++    NAMESPACE unofficial::nethost::
++)
```
It seems that the diff is caused by the suffix, please revert to LF.

julianxhokaxhiu replied:
I had to update the patch file because the `CmakeLists.txt` on the `v7.0.0` is a bit different than the one before, and [there's a new block at the end](https://github.com/dotnet/runtime/blob/main/src/native/corehost/nethost/CMakeLists.txt#L41-L49). Additionally the file is already using `LF` and was generated using `git diff > 0001-nethost-cmakelists.patch`. What am I supposed to change here?

Adela0814 commented on ports/nethost/vcpkg.json:
```diff
@@ -1,6 +1,6 @@
+  "version": "7.0.0",
   "description": "Provides a set of APIs that can be used to host .NET Core (dotnet)",
   "homepage": "https://github.com/dotnet/runtime/tree/master/src/installer/corehost/cli/nethost",
   "supports": "!uwp",
```
Please add license for nethost.

julianxhokaxhiu replied:
Added. Thanks

Adela0814 commented:
Please get [failure logs here](https://dev.azure.com/vcpkg/public/_build/results?buildId=80624&view=artifacts&pathAsName=false&type=publishedArtifacts). Please ping me if you need any help.

```
CMake Error at CMakeLists.txt:63 (target_compile_options):
  Cannot specify compile options for target "libnethost" which is not built
  by this project.
```

julianxhokaxhiu commented:
Thanks for the hint on checking the error logs on the PR. I fixed all the windows related ones, but the ones on Linux and Mac fail, although I'm not able to understand why or what is going on. Any hint is appreciated. Thanks for reviewing :)

Adela0814 commented:
I will try to repro and fix the following issue:

[config-x64-linux-out.log](https://github.com/microsoft/vcpkg/files/10055498/config-x64-linux-out.log)

julianxhokaxhiu commented:
Hi @Adela0814 any update on this PR?

Adela0814 commented:
> Hi @Adela0814 any update on this PR?

I'm sorry I haven't had time to fix this recently. I'll find time to check it as soon as possible.

Adela0814 commented:
Sorry, I can't solve this problem at the moment, I suggest convert this PR into draft. You can also try to seek help from upstream.

julianxhokaxhiu commented:
Let's see if rebasing to master helps, thanks!

julianxhokaxhiu committed:
[nethost] Update versions

julianxhokaxhiu commented:
@Adela0814 I managed to find what the issue was, apparently the project has a custom way to provide a special file included only when the build is static ( hit only by Linux and OSX targets ). I fixed that and now it builds fine under all platforms.

dg0yt commented on ports/nethost/portfile.cmake:
```diff
@@ -1,23 +1,25 @@
 if(VCPKG_TARGET_IS_WINDOWS)
   set(RID_PLAT "win")
 elseif(VCPKG_TARGET_IS_OSX)
   set(RID_PLAT "osx")
-elseif(VCPKG_TARGET_IS_LINUX)
+  execute_process(COMMAND sh -c "${SOURCE_PATH}/eng/native/version/copy_version_files.sh")
+  elseif(VCPKG_TARGET_IS_LINUX)
   set(RID_PLAT "linux")
+  execute_process(COMMAND sh -c "${SOURCE_PATH}/eng/native/version/copy_version_files.sh")
 else()
   message(FATAL_ERROR "Unsupported platform")
 endif()
```
The list of supported systems actually belongs into `vcpkg.json`:
~~~json
"supports": "(windows & !uwp), linux, osx"
~~~
Then the non-windows part can be collapsed into a single `else()` branch.

julianxhokaxhiu replied:
I'm not sure why that var is set for, as I inherited this portfile from the previous version, but I'll check and optimize as you suggested if possible. Thanks

julianxhokaxhiu replied:
I put the execute_command on a different block since the `RID_PLAT` is later used when configuring the cmake build. Let me know if this suits better.

dg0yt commented on ports/nethost/0001-nethost-cmakelists.patch:
```diff
@@ -1,73 +1,94 @@
+             echo "$version_file_contents" > "$version_file_destination"
+         fi
+     elif [[ ! -e "$__RepoRoot/artifacts/obj/$(basename "$path")" ]]; then
++        mkdir -p "$__RepoRoot/artifacts/obj/"
```
Could this be done in the portfile instead of patching? (It doesn't really belong in patch called `...cmakelists.patch`.)

julianxhokaxhiu replied:
Sure, will do, thanks!

julianxhokaxhiu replied:
Removed the patch from here and I put it on the `portfile.cmake`

julianxhokaxhiu committed:
[nethost] Revision changes

- 0001-nethost-cmakelists.patch: Remove the shell file patch and keep only the cmake patch
- portfile.cmake: Bring the execute_process entry as a single block and only if not Windows
- vcpkg.json: Extend the supports string to linux and osx

julianxhokaxhiu committed:
[nethost] Update versions

Adela0814 commented on ports/nethost/vcpkg.json:
```diff
@@ -1,9 +1,10 @@
-  "supports": "!uwp",
+  "homepage": "https://github.com/dotnet/runtime/tree/main/src/native/corehost/nethost",
+  "license": "MIT",
+  "supports": "(windows & !uwp), (linux, osx)",
```
Can you explain why `linux` and `osx` were added?

julianxhokaxhiu replied:
Nethost is a library used to load a .NET process inside an unmanaged process, and the code officially supports the scenario on all of them. So it made sense to me to add both linux and osx.

Also see the previous review comments.

dg0yt replied:
> Can you explain why `linux` and `osx` were added?

They were included in `supports` before. What is now excluded is what the portfile used to raise errors for: android, ios, freebsd etc.
